sqlglot/dialects/clickhouse.py

```diff
@@ -344,7 +344,7 @@ def safeconcat_sql(self, expression: exp.SafeConcat) -> str:
             return self.func(
                 "CONCAT",
                 *[
-                    exp.func("if", e.is_(exp.null()), e, exp.cast(e, "text"))
+                    exp.func("if", exp.Is(this=e, expression=exp.null()), e, exp.cast(e, "text"))
```

Comment:
This wasn't picked up by mypy because the expressions property is untyped, but doing e.is_(..) in this context is not correct because the method is only defined for Conditions and these arguments could be anything.

Reply:
you can't safe contact an identifier, e should always be a column

Reply:
Ok I reverted this change and just made `Placeholder`, `Parameter` and `ParameterSession` be `Condition`s.